tests/components/onvif/test_config_flow.py

```diff
@@ -0,0 +1,508 @@
+
+    with patch(
+        "homeassistant.components.onvif.config_flow.get_device"
+    ) as mock_device, patch(
+        "homeassistant.components.onvif.config_flow.wsdiscovery"
+    ) as mock_discovery, patch(
+        "homeassistant.components.onvif.camera.ONVIFHassCamera"
+    ) as mock_camera:
+        setup_mock_onvif_device(mock_device)
+        setup_mock_discovery(mock_discovery)
+        setup_mock_camera(mock_camera)
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input={}
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+        assert result["step_id"] == "device"
+        assert len(result["data_schema"].schema[config_flow.CONF_HOST].container) == 3
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input={config_flow.CONF_HOST: f"{URN} ({HOST})"}
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+        assert result["step_id"] == "auth"
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input={
+                config_flow.CONF_USERNAME: USERNAME,
+                config_flow.CONF_PASSWORD: PASSWORD,
+            },
+        )
+
+        await hass.async_block_till_done()
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+        assert result["title"] == f"{URN} - {MAC}"
+        assert result["data"] == {
+            config_flow.CONF_NAME: URN,
+            config_flow.CONF_HOST: HOST,
+            config_flow.CONF_PORT: PORT,
+            config_flow.CONF_USERNAME: USERNAME,
+            config_flow.CONF_PASSWORD: PASSWORD,
+            config_flow.CONF_PROFILE: [0],
+        }
+
+
+async def test_flow_discovered_devices_ignore_configured_manual_input(hass):
+    """Test that config flow discovery ignores configured devices."""
+    await setup_onvif_integration(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        config_flow.DOMAIN, context={"source": "user"}
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+
+    with patch(
+        "homeassistant.components.onvif.config_flow.get_device"
+    ) as mock_device, patch(
+        "homeassistant.components.onvif.config_flow.wsdiscovery"
+    ) as mock_discovery, patch(
+        "homeassistant.components.onvif.camera.ONVIFHassCamera"
+    ) as mock_camera:
+        setup_mock_onvif_device(mock_device)
+        setup_mock_discovery(mock_discovery, with_mac=True)
+        setup_mock_camera(mock_camera)
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input={}
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+        assert result["step_id"] == "device"
+        assert len(result["data_schema"].schema[config_flow.CONF_HOST].container) == 2
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5453535)